### PR TITLE
Upgrade fastify to v3

### PR DESCRIPTION
**Pull request by frikille**
* Updates fastify dependency to v3
* Update type definitions
* Move type tests to tsd
* Update minimum version string
* Add Node v14 to test matrix
* Move from Azure pipelines to Github actions
* Add dependabot integration

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+version: 1
+update_configs:
+  - package_manager: "javascript"
+    directory: "/javascript"
```
Comment (mcollina) [inline .dependabot/config.yml:4]:
```suggestion
    directory: "/"
```

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "Automatically generate etags for HTTP responses, for Fastify",
   "main": "index.js",
   "scripts": {
-    "typescript": "tsc --project ./test/types/tsconfig.json",
+    "typescript": "tsd",
     "test": "standard | snazzy && tap -J test/*.test.js && npm run typescript",
```
Comment (Eomm) [inline package.json:8]:
```suggestion
    "test": "standard | snazzy && tsd && tap -J test/*.test.js && npm run typescript",
```

File: test/types/index.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import * as fastify from "fastify";
-import * as fastifyEtag from "../../";
+import fastify from "fastify";
+import fastifyEtag from "../../";
```
Comment (mcollina) [inline test/types/index.ts:2]: Can you add some assertions with tsd?

File: index.js [hunk 4]
```diff
@@ -39,6 +39,6 @@ module.exports = fp(async function etag (app, opts) {
     done(null, newPayload)
   })
 }, {
-  fastify: '2.x',
+  fastify: '>=3',
```
Comment (mcollina) [inline index.js:42]:
```suggestion
  fastify: '3.x',
```

**Pull request merged**